### DISCUSSION TITLE
CustomMenu with individual key bindings

### DISCUSSION THREAD

**Discussion by mortensen** [AlmasB/FXGL#1103] [Q&A]
If you dont want the "default" menu you can override it and add a custom menu with custom backgrounds and buttons. It extends FXGLMenu. But the types are fix (GameMenu and MainMenu). And you dont seem to be able to change the key bindings for opening and closing the Ingame menu.

Use case:
Have a game open - RPG. Be in a Vendor UI. When you now press ESC, the vendor UI gets closed. But also the menu gets opened.

I would like to be able to add custom code to the key bindings and say, if UIManager.myWindowOpen... -> dont open GameMenu. Else dont open it.

Is this a new feature? Or possible with FXGL?

**Comment by AlmasB** [comment]
You can change the default menu key: `settings.setMenuKey( ... )` (or something along these lines)

**Reply by mortensen** [reply]
Ok but I think ESC is quite common. It's just that it collides with other menus that may be open . Isn't there a way to make an overridable method in FXGLMenu called fireMenuOpen and fireMenuClosed or something like this ? :) Where I can then decide with a boolean return value wether the menu should be opened or not.

**Reply by AlmasB** [reply]
You could try adding an input binding to ESC, so you can intercept the menu key calls before FXGL and then decide whether to open the menu or not. You may need to set the menu key to be something different for this to work.

Also note that each SubScene has its own `Input` object that gets input event calls, may or may not be helpful.
